pallets/pallet-bonded-coins/src/lib.rs

```diff
@@ -681,12 +688,35 @@ pub mod pallet {
 					sum.saturating_add(T::Fungibles::total_issuance(id))
 				});
 
-			let amount = burnt
-				.checked_mul(&total_collateral_issuance)
-				.ok_or(ArithmeticError::Overflow)? // TODO: do we need a fallback if this fails?
+			defensive_assert!(
+				sum_of_issuances >= burnt,
+				"burnt amount exceeds the total supply of all bonded currencies"
+			);
+
+			// divide first to avoid overflow, remainder will be handled seperately
+			let divisible_part = total_collateral_issuance
+				.checked_div(&sum_of_issuances)
+				.ok_or(Error::<T>::Internal)? // because sum_of_issuances >= burnt > 0, this is theoretically impossible
+				.checked_mul(&burnt)
+				// An overflow is also theoretically impossible, as burnt <= sum_of_issuances
+				.ok_or(Error::<T>::Internal)?;
```

Comment:
`Internal` errors without an associated log is pretty much useless. Not sure if logging is supposed to be part of a separate PR or not, so I'll just mention it here.

Reply:
I've got the `defensive_assert!` which should do some logging in production builds, but other than that we haven't really set up logging yet.

Reply:
Ok then this is a thing to tackle when logging is added, which should happen at the very least any time an `Internal` error is thrown.